AdyenComponents/Apple Pay/ApplePayComponentExtensions.swift

```diff
@@ -34,14 +34,114 @@ extension ApplePayComponent: PKPaymentAuthorizationViewControllerDelegate {
+                                   for: applePayPayment,
+                                   with: { [weak self] result in
+                                       guard let self = self else {
+                                           return completion(.init(paymentSummaryItems: applePayPayment.summaryItems))
+                                       }
+
+                                       if result.status == .success,
+                                          result.paymentSummaryItems.isEmpty == false {
+                                           do {
+                                               try self.applePayPayment.update(with: result.paymentSummaryItems)
+                                           } catch {
+                                               self.delegate?.didFail(with: error, from: self)
+                                           }
+                                       }
+                                       completion(result)
+                                   })
+    }
+
+    @available(iOS 15.0, *)
+    public func paymentAuthorizationViewController(_ controller: PKPaymentAuthorizationViewController,
+                                                   didChangeCouponCode couponCode: String,
+                                                   handler completion: @escaping (PKPaymentRequestCouponCodeUpdate) -> Void) {
+        guard let applePayDelegate = applePayDelegate else {
+            return completion(.init(paymentSummaryItems: applePayPayment.summaryItems))
+        }
+
+        let applePayPayment = self.applePayPayment
+        applePayDelegate.didUpdate(couponCode: couponCode,
+                                   for: applePayPayment,
+                                   with: { [weak self] result in
+                                       guard let self = self else {
+                                           return completion(.init(paymentSummaryItems: applePayPayment.summaryItems))
+                                       }
+            
+                                       if result.status == .success,
+                                          result.paymentSummaryItems.isEmpty == false {
+                                           do {
+                                               try self.applePayPayment.update(with: result.paymentSummaryItems)
+                                           } catch {
+                                               self.delegate?.didFail(with: error, from: self)
+                                           }
+                                       }
+                                       completion(result)
+                                   })
+    }
+}
+
+public protocol ApplePayComponentDelegate: AnyObject {
+
+    func didUpdate(contact: PKContact,
+                   for payment: ApplePayPayment,
+                   with completion: @escaping (PKPaymentRequestShippingContactUpdate) -> Void)
```

Comment:
Maybe personal but not a fan of `with` for the completion handler :)
It usually is just `completion`